classes/submission/Repository.inc.php

```diff
@@ -99,11 +100,21 @@ public function createDois(Submission $submission): void
 
         // Article
         $publication = $submission->getCurrentPublication();
-        if ($context->isDoiTypeEnabled(Repo::doi()::TYPE_PUBLICATION) && empty($publication->getData('doiId'))) {
-            $doiId = Repo::doi()->mintPublicationDoi($publication, $submission, $context);
-            if ($doiId !== null) {
-                Repo::publication()->edit($publication, ['doiId' => $doiId]);
+
+        $doiCreationFailures = [];
+
+        // TODO: #dois Should we also throw a DoiCreationException if a DOI has already been assigned indicating
+        //  as such or should it "fail" to create a new DOI silently
+        try {
+            if ($context->isDoiTypeEnabled(Repo::doi()::TYPE_PUBLICATION) && empty($publication->getData('doiId'))) {
+                $doiId = Repo::doi()->mintPublicationDoi($publication, $submission, $context);
+                if ($doiId !== null) {
```

Comment:
I'm noticing that here (and in all `mintXXXXXDoi()` methods) you're checking for a `null` result, and the method is type hinted with `?int`. I'm not actually sure what the correct thing is here, but I'm pretty sure that if an exception is thrown, line 111 will never run. It immediately jumps to the `catch` block.

I'm going to ask in slack about the correct way to handle this. But I think you can remove the null check everywhere you've got it in all the apps.